zoo/services/assets/js/service_form.js

```diff
@@ -44,15 +53,25 @@ $("button.remove-environment").click(function () {
 
 $("button.add-environment").click(function () {
     $(this).blur();
-    var hidden = $(".ui.segment.environment:hidden");
+    let hidden = $(".ui.segment.environment:hidden");
     if (hidden.length > 0) {
         if (hidden.length == 1) {
             $(this).prop("disabled", true);
         }
 
-        var firstSegment = hidden.eq(0);
+        let firstSegment = hidden.eq(0);
         firstSegment.find(":not(.no-reset) input").val("");
         firstSegment.find(".field.name input").prop("required", true);
+        firstSegment.find('input[name$="type"]').val(envTypeZoo);
         firstSegment.show();
     }
 });
+
+$(".load-gitlab-environment").on("click", function () {
```

Comment:
hmm... what about firing this whenever the repository field is updated on the service form instead? what do you think?

Reply:
I have removed button. Now envs are loaded on init and on changed of repo